pkg/jobs/job_scheduler.go

```diff
@@ -547,6 +548,11 @@ func StartJobSchedulerDaemon(
 	cfg *scheduledjobs.JobExecutionConfig,
 	env scheduledjobs.JobSchedulerEnv,
 ) {
+	// Since the job scheduler system is not under user control, exclude it from
+	// from cost accounting and control. Individual jobs are not part of this
+	// exclusion.
+	ctx = multitenant.WithTenantCostControlExemption(ctx)
```

Comment:
If I understand this mechanism correctly, any action performed with context with cost control exemption marker will be excluded from cost accounting.

It looks like jobs system ensures that's the case by verify if the job (resumer) marked itself as exempt.  However, the way scheduled jobs are written ATM, there is nothing that requires a schedule implementation to spawn a job.  What the schedule does -- is up to the schedule implementation. At this time -- all schedule implementations do spawn jobs.  But I wonder if we should be defensive and clear out exemption marker right before we execute the schedule (https://github.com/cockroachdb/cockroach/blob/master/pkg/jobs/job_scheduler.go#L184-L184).

Reply:
I did have it that way originally, but it was more complex because I needed to add a "don't exempt" value to the context. I want to be able to back-port this, and it was getting rather complex, so I backed off and stuck with what you see. I'm not super-worried about job schedulers that don't schedule jobs - we just need to make sure we're charging for the 99% cases, and not get too hung up on the 1% case.